Sources/MapboxDirections/Extensions/ForeignMemberContainer.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+import Turf
+
+/**
+ A coding key as an extensible enumeration.
+ */
+struct AnyCodingKey: CodingKey {
+    var stringValue: String
+    var intValue: Int?
+
+    init?(stringValue: String) {
+        self.stringValue = stringValue
+        self.intValue = nil
+    }
+
+    init?(intValue: Int) {
+        self.stringValue = String(intValue)
+        self.intValue = intValue
+    }
+}
+
+extension ForeignMemberContainer {
+    /**
+     Decodes any foreign members using the given decoder.
+     */
+    mutating func decodeForeignMembers<WellKnownCodingKeys>(notKeyedBy _: WellKnownCodingKeys.Type, with decoder: Decoder) throws where WellKnownCodingKeys: CodingKey {
```

Comment:
Turf didn't expose these methods. It doesn't break existing conformances, but if we want to use it in Directions, we need these methods.
Should `ForeignMemberClassContainer` follow the same rule? I feel that it's `decoding` method should remain public to allow users successfully subclass our `open` types. Otherwise, the mechanism will not handle new subclass properties and put them into `foreignMembers` as unrecognized.

Reply:
I think that’s reasonable. Turf’s structs can’t be subclassed, but MapboxDirections’ classes are often subclassed, making it more important to provide conveniences like this method.

Reply:
>Is ForeignMemberClassContainer only separate from ForeignMemberContaining because of the need to make these methods open rather than public?

No, the difference is caused by value and reference types difference. `ForeignMemberContainer` uses `mutating` method which is fine to use in struct's `init`, but classes do not allow it since `self` is immutable during initialization methods (such as `init(from decoder)`. 
Also, encoding is different too. Method should be called only in the base class and since structs does not care about subclassing it can use provided `CodingKeys`, while class implementation has to rely on `foreignMembers` to be correctly populated beforehand.
Given that I've decided to put it into another protocol.